packages/parser/lib/css_parser.mly

```diff
@@ -89,29 +89,131 @@ bracket_block (X): xs = delimited(LEFT_BRACKET, X, RIGHT_BRACKET) { xs }
 paren_block (X): xs = delimited(LEFT_PAREN, X, RIGHT_PAREN) { xs }
 
 /* https://www.w3.org/TR/mediaqueries-5 */
-/* Parsing with this approach is almost as good the entire spec */
 
-prelude: xs = loption(nonempty_list(loc(value_in_prelude))) { xs }
-
-/* Missing grammars: */
-/* (width >= 600px) */
-/* (400px < width < 1000px) */
-/* (not (color)) and (not (hover)) */
-/* Combinator "," */
-media_query_prelude_item:
-  | i = IDENT { Ident i }
+// TODO: Alot of duplication here, any way to improve it?
+/* ">" */
+greater_operator: o = OPERATOR { o }
+/* "<" */
+less_operator: o = OPERATOR { o }
+/* "=" */
+equal_sign: e = OPERATOR { e }
+/* "and" */
+and_operator: o = OPERATOR { o }
+/* "or" */
+or_operator: o = OPERATOR { o }
+/* "only" */
+only_operator: o = OPERATOR { o }
+/* "not" */
+not: n = OPERATOR { n }
+/* "all" | "screen" | "print" */
+media_type: s = IDENT { s }
+
+/* https://www.w3.org/TR/mediaqueries-5/#mq-syntax */
+
+/* <mf-plain> = <mf-name> : <mf-value> */
```

Comment:
Would define a certain list of mf-name make the parsing better here?

Reply:
Absolutely, it would be better if we could somehow call the `Reason` parser's `check_property` somewhere. I'm not sure yet how this will evolve.